### PR TITLE
Identity service

### PR DESCRIPTION
**Pull request by Hostileca**

### CODE REVIEW

File: Backend/IdentityService/BusinessLogicLayer/Services/Interfaces/IAuthService.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using BusinessLogicLayer.Result;
+using BusinessLogicLayer.ViewModels;
+
+namespace BusinessLogicLayer.Services.Interfaces;
+
+public interface IAuthService
+{
+    Task<Results> RegisterAsync(RegisterViewModel registerViewModel);
+    Task<Results> LoginAsync(LoginViewModel loginViewModel);
```
Comment (MefistoSamael) [inline Backend/IdentityService/BusinessLogicLayer/Services/Interfaces/IAuthService.cs:9]: empty lines between methods

File: Backend/IdentityService/BusinessLogicLayer/BusinessLogicLayer.csproj [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>net8.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <Nullable>enable</Nullable>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <ProjectReference Include="..\DataAccessLayer\DataAccessLayer.csproj" />
```
Comment (MefistoSamael) [inline Backend/IdentityService/BusinessLogicLayer/BusinessLogicLayer.csproj:10]: business logic layer must not to have dependencies on DAL

File: Backend/IdentityService/DataAccessLayer/Models/User.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿using Microsoft.AspNetCore.Identity;
+
+namespace DataAccessLayer.Models;
+
+public class User : IdentityUser
+{
+}
```
Comment (MefistoSamael) [inline Backend/IdentityService/DataAccessLayer/Models/User.cs:7]: place user in bll

**Review by MefistoSamael — CHANGES_REQUESTED** [review]

File: Backend/IdentityService/BusinessLogicLayer/BusinessLogicLayer.csproj (hunk 2 above)
Comment (Hostileca) [reply Backend/IdentityService/BusinessLogicLayer/BusinessLogicLayer.csproj:10]: So, if I place models in bll, DAL will have dependencies on BL. Is it correct?

File: Backend/IdentityService/BusinessLogicLayer/Exceptions/DuplicateException.cs
Comment (MefistoSamael) [inline Backend/IdentityService/BusinessLogicLayer/Exceptions/DuplicateException.cs:1]: rename it to alreadyexists

File: Backend/IdentityService/BusinessLogicLayer/Exceptions/AuthorizationException.cs
Comment (MefistoSamael) [inline Backend/IdentityService/BusinessLogicLayer/Exceptions/AuthorizationException.cs:1]: rename to unauthorized

File: Backend/IdentityService/BusinessLogicLayer/Services/Implementations/AccountService.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+{
+    public async Task<UserReadDto> RegisterAsync(UserRegisterDto userRegisterDto)
+    {
+        var user = await userManager.FindByEmailAsync(userRegisterDto.Email);
+        if (user is not null)
+        {
+            throw new DuplicateException(typeof(User).ToString());
+        }
+        
+        var newUser = new User { UserName = userRegisterDto.Username, Email = userRegisterDto.Email };
```
Comment (MefistoSamael) [inline Backend/IdentityService/BusinessLogicLayer/Services/Implementations/AccountService.cs:22]: use mapper

File: Backend/IdentityService/BusinessLogicLayer/Services/Implementations/AccountService.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        var result = await userManager.CreateAsync(newUser, userRegisterDto.Password);
+        
+        if (!result.Succeeded)
+        {
+            throw new AuthorizationException("User creation failed");
```
Comment (MefistoSamael) [inline Backend/IdentityService/BusinessLogicLayer/Services/Implementations/AccountService.cs:27]: are you sure that it is unauthorized?

File: Backend/IdentityService/BusinessLogicLayer/Services/Implementations/AccountService.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    UserManager<User> userManager) 
+    : IAccountService
+{
+    public async Task<UserReadDto> RegisterAsync(UserRegisterDto userRegisterDto)
+    {
+        var user = await userManager.FindByEmailAsync(userRegisterDto.Email);
+        if (user is not null)
+        {
+            throw new DuplicateException(typeof(User).ToString());
+        }
```
Comment (MefistoSamael) [inline Backend/IdentityService/BusinessLogicLayer/Services/Implementations/AccountService.cs:20]: all conditional statements must me preceeded and followed by blank lines

File: Backend/IdentityService/DataAccessLayer/Models/User.cs
Comment (MefistoSamael) [inline Backend/IdentityService/DataAccessLayer/Models/User.cs:1]: move to business logic layer

Comment (MefistoSamael) [inline Backend/IdentityService/DataAccessLayer/Models/User.cs:1]: move to business logic layer

**Review by MefistoSamael — CHANGES_REQUESTED** [review]
Add jwt/refresh tokens. Add user crud

**Review by MefistoSamael — APPROVED** [review]

**Pull request merged**